### PR TITLE
fix(macOS): Vertically arranged window control buttons not work

### LINKED ISSUE #105: macOS: The vertically arranged window control buttons only have the first button functional
**Describe the bug**
The vertically arranged window control buttons on macOS only have the first button functional.

**To Reproduce**
Just open the app.

Safari浏览器 2024-11-23 00 32 14

The yellow and green buttons are not working.

The area within the red box is the actual area that responds to hover, but it cannot respond to clicks either.

**Expected behavior**
They should work...

### PR DESCRIPTION
**Pull request by XMLHexagram**
fix #105

CleanShot 2024-11-27 at 02 51 22@2x

## Summary by Sourcery

Bug Fixes:
- Fix the issue with vertically arranged window control buttons not working on macOS by implementing a method to set their positions.

### CODE REVIEW

File: macos/Runner/MainFlutterWindow.swift [hunk 1]
```diff
@@ -17,8 +18,68 @@ class MainFlutterWindow: BitsdojoWindow {
...
+  func setVertical() {
+    overrideStandardWindowButtonPosition(buttonType: .closeButton, offset: .init(x: 8, y: 8))
+    overrideStandardWindowButtonPosition(buttonType: .miniaturizeButton, offset: .init(x: 8, y: 28))
+    overrideStandardWindowButtonPosition(buttonType: .zoomButton, offset: .init(x: 8, y: 48))
+  }
+
+  func overrideStandardWindowButtonPosition(
+    buttonType: NSWindow.ButtonType, offset: CGPoint
+  ) {
+    guard let standardWindowButton = mainFlutterWindow!.standardWindowButton(buttonType) else {
```
[the remark this answers is not in the document]
Comment (XMLHexagram) [reply macos/Runner/MainFlutterWindow.swift:61]: not a problem

File: macos/Runner/MainFlutterWindow.swift [hunk 2]
```diff
@@ -17,8 +18,68 @@ class MainFlutterWindow: BitsdojoWindow {
...
+      return
+    }
+
+    standardWindowButton.removeFromSuperview()
+
+    guard let contentView: NSView = mainFlutterWindow!.contentView else {
```
[the remark this answers is not in the document]
Comment (XMLHexagram) [reply macos/Runner/MainFlutterWindow.swift:67]: not a problem

File: lib/main.dart [hunk 3]
```diff
@@ -6,6 +6,7 @@ import 'package:provider/provider.dart';
 import 'package:flutter/foundation.dart';
 import 'package:fluent_ui/fluent_ui.dart';
 import 'package:intl/intl_standalone.dart';
+import 'package:rune/utils/macos_window_control_button_manager.dart';
```
Comment (Losses) [inline lib/main.dart:9]: Please follow the code convention, files in the project should be imported relatively and sorted.

File: macos/Runner/MainFlutterWindow.swift [hunk 4]
```diff
@@ -1,7 +1,8 @@
+import AppKit
 import Cocoa
 import FlutterMacOS
-import macos_window_utils
 import bitsdojo_window_macos
+import macos_window_utils
```
Comment (Losses) [inline macos/Runner/MainFlutterWindow.swift:5]: Is macos_window_utils still necessary for this project?

**Review by Losses — CHANGES_REQUESTED** [review]
Please update the code as the review note

File: lib/main.dart (hunk 3 above)
Comment (Losses) [reply lib/main.dart:9]: You are not fixing this.

**Review by Losses — CHANGES_REQUESTED** [review]

File: lib/main.dart (hunk 3 above)
Comment (XMLHexagram) [reply lib/main.dart]: Sorry I missed this.

File: lib/main.dart [hunk 5]
```diff
@@ -12,8 +12,7 @@ import 'package:flutter_acrylic/flutter_acrylic.dart';
 import 'package:bitsdojo_window/bitsdojo_window.dart';
 import 'package:device_info_plus/device_info_plus.dart';
 import 'package:flutter_fullscreen/flutter_fullscreen.dart';
-import 'package:macos_window_utils/macos_window_utils.dart';
-import 'package:macos_window_utils/macos/ns_window_button_type.dart';
+import 'package:rune/utils/macos_window_control_button_manager.dart';
```
Comment (Losses) [inline lib/main.dart:15]: Please follow this convention

https://github.com/Losses/rune/blob/0f77ae66301976a84c4dd04bf2e401506dcfa8e5/lib/main.dart#L19-L32

**Review by Losses — CHANGES_REQUESTED** [review]

File: lib/main.dart (hunk 5 above)
Comment (Losses) [reply lib/main.dart:15]: ¯\_(ツ)_/¯

**Pull request merged**